routes/markers.js

```diff
@@ -141,72 +141,80 @@ module.exports = function(connection) {
     /* Update marker with given id */
     router.patch('/marker/:id', utils.isAuthenticated, function (req, res, next) {
 
-        //If authenticated user is not the owner of the marker or an admin, return 401 (Unauthorized)
-        if(req.session.user.uid != req.body.uid && !req.session.user.admin) return res.send(401);
-
-        if(req.body !== null && !Object.keys(req.body).length) {
-            var id = req.params['id']
-            var date_changed=new Date().toISOString().slice(0, 19).replace('T', ' ');
-            var query = 'UPDATE atlas SET date_changed=? WHERE id =?';
-            if(!req.session.user.admin) {
-                query += ' AND created_by=\''+req.session.user.uid+'\'';
-            }
+        var id = req.params['id']
 
-            connection.query(query, [date_changed,id], function (error, rows,field) {
-                if(!!error){
-                    console.log(error);
-                }
-            });            
+        connection.query("SELECT created_by FROM atlas WHERE id=?", [id], function (error, rows, field) {
+            if(error) {
+                console.log(error);
```

Comment:
We should probably be returning a proper server error response to the client like
```node
return res.status(500).send({ message: "Error retrieving created_by from database"});
```
after logging an error to the server.

Reply:
Updated the PR. Server errors are barely being handled in our server code, we should probably create a ticket for that.